contracts/JoeBarV2.sol

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.6.12;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/math/SafeMathUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/proxy/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+interface IBarRewarder {
+    function claimReward() external;
+}
+
+// JoeBar is the coolest bar in town. You come in with some Joe, and leave with more! The longer you stay, the more Joe you get.
+//
+// This contract handles swapping to and from moJoe, JoeSwap's staking token.
+contract JoeBarV2 is Initializable, ERC20Upgradeable, OwnableUpgradeable {
+    using SafeMathUpgradeable for uint256;
+
+    IERC20Upgradeable public joe;
+    IBarRewarder public rewarder;
+    uint256 entryFee;
+
+    event SetEntryFee(uint256 _entryFee);
+    event SetRewarder(address _rewarder);
+
+    // Define the Joe token contract
+    function initialize(IERC20Upgradeable _joe, uint256 _entryFee) public initializer {
+        require(_entryFee <= 5000, "JoeBarV2: entryFee too high");
+        __ERC20_init("JoeBarV2", "moJOE");
+        __Ownable_init();
+
+        joe = _joe;
+        entryFee = _entryFee; // in basis points aka parts per 10,000 so 5000 is 50%, entry fee of 50%.
```

Comment:
You should probably use the `setEntryFee` function here as setting directly will skip the `require(_entryFee <= 5000, "JoeBarV2: entryFee too high");` check.